apps/condo/domains/billing/schema/RegisterBillingReceiptsService.js

```diff
@@ -334,13 +334,16 @@ const RegisterBillingReceiptsService = new GQLCustomSchema('RegisterBillingRecei
 
                 // Step 1:
                 // Parse properties, accounts and receipts from input
+                const categoryResolver = new CategoryResolver()
```

Comment:
why use class here ?

Reply:
Because, there will be a single class for every instance the unified mutation will work: Category, Account, Recipient, Property, Receipt. They all will have base class for logging, error handling and caching mechanics. Work with receipt will be like using the chain of classes. And yes, they will be singleton . To separate logic to different classes is better then to have one super-function 5 pages in length. It will be also easy to write tests for them as they have predictable input and needs to produce predictable output.